bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java

```diff
@@ -1693,7 +1693,12 @@ private static boolean checkObjectEquivalency(Object sourceVal, Type sourceType,
     }
     
     private static List<MethodType> getAllFunctionsList(BObjectType objectType) {
-        List<MethodType> functionList = new ArrayList<>(Arrays.asList(objectType.getMethods()));
+        List<MethodType> functionList = new ArrayList<>();
+        for (MethodType method : objectType.getMethods()) {
+            if (!method.getName().equals("init")) {
+                functionList.add(method);
```

Comment:
this will have impact on object equal check performance. Need to think different way to handle this